### PR TITLE
DDF-5194 Refactors metacard-interactions to be extensible via extension points

### PR DESCRIPTION
**Pull request by cantstoptheunk**
### What does this PR do?
Currently metacard-interactions is extensible via plugins, it would be nice to move to the new style of extensibility (extension points) without breaking code bases that use plugins

#### Who is reviewing it? 
@adimka 
@Bdthomson 
@andrewkfiedler 
@djblue 
@vinamartin 

#### Ask 2 committers to review/merge the PR and tag them here.
@adimka
@andrewkfiedler
@djblue
@vinamartin

#### How should this be tested?
In a downstream project, make sure the metacard-interaction options still work. 
Screen Shot 2019-08-12 at 3 42 41 PM

#### Any background context you want to provide?
Used `https://github.com/codice/ddf/pull/5060` as a guide

#### What are the relevant tickets?
Fixes: #5194 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-interactions/index.tsx [hunk 1]
```diff
@@ -16,16 +16,8 @@ import * as React from 'react'
...
-import CreateLocationSearch from './location-interaction'
-import ExpandMetacard from './expand-interaction'
-import BlacklistToggle, { isBlacklisted } from './hide-interaction'
-import DownloadProduct from './download-interaction'
-import ExportActions from './export-interaction'
-import AddToList from './add-to-list-interaction'
-import { Divider } from '../../presentation/metacard-interactions/metacard-interactions'
-
-const plugin = require('plugins/metacard-interactions')
+import { isBlacklisted } from './hide-interaction'
```
Comment (cantstoptheunk) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-interactions/index.tsx]: This is used on line 59 `blacklisted: isBlacklisted(props.model),`

**Comment by cantstoptheunk** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/extension-points/extension-points.tsx [hunk 2]
```diff
@@ -33,6 +34,7 @@ export type ExtensionPointsType = {
   visualizations: any[]
   queryForms: any[]
   navigationRight: any[]
+  metacardInteractions: any[]
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/extension-points/extension-points.tsx]: Is there a way to type these in a more helpful way, such as React.ReactNode?

Comment (cantstoptheunk) [reply ui/packages/catalog-ui-search/src/main/webapp/extension-points/extension-points.tsx]: I looked at a downstream project and they use things that are not React Components (ie. a local variable)

Comment (cantstoptheunk) [reply ui/packages/catalog-ui-search/src/main/webapp/extension-points/extension-points.tsx]: Talked @adimka, he's going to make downstream changes and I'll change this to be of type ReactNode

**Comment by Corey-Collins** [comment]
Hero successful :tada: :tada:
verified metacard-interaction options on downstream projects still work

**Comment by jlcsmith** [comment]
build now

**Comment by bellcc** [comment]
build now

**Pull request merged**